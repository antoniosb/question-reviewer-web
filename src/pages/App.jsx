import React from 'react';
import AuthPage from '../pages/Auth';
import { HashRouter as Router, Route, Switch } from "react-router-dom";
import { Layout } from 'antd';
import NotFoundPage from '../pages/NotFound';
const { Header, Content, Footer } = Layout;
import './../assets/scss/pages/app.scss'
import PrivateRoute from '../components/PrivateRoute';
import QuestionList from './QuestionList';
import QuestionCrudPage from './QuestionCrud';
import AppHeader from '../components/AppHeader';

class App extends React.PureComponent {
  render() {
    return (
      <Router>
        <Layout className="app">
          <Header className="app-header">
            <AppHeader />
          </Header>
          <Content className="app-content">
            <div>
              <Switch>
                <PrivateRoute exact path="/" component={QuestionList}/>
                <PrivateRoute exact path="/questions" component={QuestionList}/>
                <PrivateRoute exact path="/questions/:id" component={QuestionCrudPage}/>
                <Route path="/login" component={AuthPage} key="login"/>
                <Route path="/signUp" component={AuthPage} key="signUp"/>
                <Route component={NotFoundPage} />
              </Switch>
            </div>
          </Content>
          <Footer className="app-footer">
            Banco de Questões ©2018 criado por Hermógenes Ferreira
          </Footer>
        </Layout>
      </Router>
    );
  }
}

export default App
