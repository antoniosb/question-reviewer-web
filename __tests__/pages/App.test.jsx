import React from 'react'
import App from '../../src/pages/App'

describe('Pages > App', () => {
  test('Render', () => {
    const app = shallow(<App />)

    expect(app).toMatchSnapshot()
  })
})