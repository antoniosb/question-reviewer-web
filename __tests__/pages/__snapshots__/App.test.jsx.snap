// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages > App Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Adapter
        className="app"
      >
        <Adapter
          className="app-header"
        >
          <Connect(AppHeader) />
        </Adapter>
        <Adapter
          className="app-content"
        >
          <div>
            <Switch>
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/questions"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/questions/:id"
              />
              <Route
                component={[Function]}
                path="/login"
              />
              <Route
                component={[Function]}
                path="/signUp"
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </div>
        </Adapter>
        <Adapter
          className="app-footer"
        >
          Banco de Questões ©2018 criado por Hermógenes Ferreira
        </Adapter>
      </Adapter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Adapter
            className="app-header"
          >
            <Connect(AppHeader) />
          </Adapter>,
          <Adapter
            className="app-content"
          >
            <div>
              <Switch>
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions/:id"
                />
                <Route
                  component={[Function]}
                  path="/login"
                />
                <Route
                  component={[Function]}
                  path="/signUp"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </div>
          </Adapter>,
          <Adapter
            className="app-footer"
          >
            Banco de Questões ©2018 criado por Hermógenes Ferreira
          </Adapter>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(AppHeader) />,
            "className": "app-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <Switch>
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions/:id"
                />
                <Route
                  component={[Function]}
                  path="/login"
                />
                <Route
                  component={[Function]}
                  path="/signUp"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </div>,
            "className": "app-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Switch>
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions/:id"
                />
                <Route
                  component={[Function]}
                  path="/login"
                />
                <Route
                  component={[Function]}
                  path="/signUp"
                />
                <Route
                  component={[Function]}
                />
              </Switch>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/"
                  />,
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions"
                  />,
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions/:id"
                  />,
                  <Route
                    component={[Function]}
                    path="/login"
                  />,
                  <Route
                    component={[Function]}
                    path="/signUp"
                  />,
                  <Route
                    component={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/questions",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/questions/:id",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "login",
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "signUp",
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/signUp",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Banco de Questões ©2018 criado por Hermógenes Ferreira",
            "className": "app-footer",
          },
          "ref": null,
          "rendered": "Banco de Questões ©2018 criado por Hermógenes Ferreira",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Adapter
          className="app"
        >
          <Adapter
            className="app-header"
          >
            <Connect(AppHeader) />
          </Adapter>
          <Adapter
            className="app-content"
          >
            <div>
              <Switch>
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/questions/:id"
                />
                <Route
                  component={[Function]}
                  path="/login"
                />
                <Route
                  component={[Function]}
                  path="/signUp"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </div>
          </Adapter>
          <Adapter
            className="app-footer"
          >
            Banco de Questões ©2018 criado por Hermógenes Ferreira
          </Adapter>
        </Adapter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Adapter
              className="app-header"
            >
              <Connect(AppHeader) />
            </Adapter>,
            <Adapter
              className="app-content"
            >
              <div>
                <Switch>
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions"
                  />
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions/:id"
                  />
                  <Route
                    component={[Function]}
                    path="/login"
                  />
                  <Route
                    component={[Function]}
                    path="/signUp"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>
              </div>
            </Adapter>,
            <Adapter
              className="app-footer"
            >
              Banco de Questões ©2018 criado por Hermógenes Ferreira
            </Adapter>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(AppHeader) />,
              "className": "app-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <Switch>
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions"
                  />
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions/:id"
                  />
                  <Route
                    component={[Function]}
                    path="/login"
                  />
                  <Route
                    component={[Function]}
                    path="/signUp"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>
              </div>,
              "className": "app-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Switch>
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions"
                  />
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/questions/:id"
                  />
                  <Route
                    component={[Function]}
                    path="/login"
                  />
                  <Route
                    component={[Function]}
                    path="/signUp"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Connect(PrivateRoute)
                      component={[Function]}
                      exact={true}
                      path="/"
                    />,
                    <Connect(PrivateRoute)
                      component={[Function]}
                      exact={true}
                      path="/questions"
                    />,
                    <Connect(PrivateRoute)
                      component={[Function]}
                      exact={true}
                      path="/questions/:id"
                    />,
                    <Route
                      component={[Function]}
                      path="/login"
                    />,
                    <Route
                      component={[Function]}
                      path="/signUp"
                    />,
                    <Route
                      component={[Function]}
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/questions",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/questions/:id",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "login",
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/login",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "signUp",
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/signUp",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Banco de Questões ©2018 criado por Hermógenes Ferreira",
              "className": "app-footer",
            },
            "ref": null,
            "rendered": "Banco de Questões ©2018 criado por Hermógenes Ferreira",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
